src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2877,6 +2874,12 @@ void InitSolarHeatGains(EnergyPlusData &state)
                     state.dataHeatBal->SurfQRadSWOutIncBmToDiffReflGnd(SurfNum) + state.dataHeatBal->SurfQRadSWOutIncSkyDiffReflGnd(SurfNum);
             }
         }
+
+        Array1D<Real64> currCosInc(state.dataSurface->TotSurfaces);      // Cosine of incidence angle of beam solar on glass
+        Array1D<Real64> currBeamSolar(state.dataSurface->TotSurfaces);   // Local variable for BeamSolarRad
+        Array1D<Real64> currSkySolarInc(state.dataSurface->TotSurfaces); // Sky diffuse solar incident on a surface
+        Array1D<Real64> currGndSolarInc(state.dataSurface->TotSurfaces); // Ground diffuse solar incident on a surface
```

Comment:
Are these happening every timestep? These should be moved out of here. That can be later since this is just moving them down in the same function.

Reply:
Yeah, I moved these so they now only get allocated if the sun is up, but I think we can probably figure out a way to remove them entirely on a separate branch.

Reply:
Or at the very least put the into `state.dataSurface`. Objects with heap allocation should not be used as local variables.

Reply:
So I guess you are intending on leaving this for now?  It would be nice to get these cleaned up.  I'm still not sure I'm ready to hold this up for it, but this allocation occurring regularly seems pretty problematic.  I may be missing something from context.

Reply:
I guess a point is that this is not actually _adding_ these new variables, it's moving them into this block, which is actually an improvement.  I think.  So that makes me feel better, but it should still be cleaned up.